### ISSUE TITLE
No failure when used as pre-commit hook

### ISSUE THREAD

**Issue by yajo** [KyleKing/copier-template-tester#3]
**Describe the bug**

I used this as a pre-commit hook. The hook generated new files. Git commit passed OK, instead of failing.

#**To Reproduce**

Steps and/or code snippet(s) to reproduce the behavior:

1. Add a new output.
1. Configure the pre-commit hook.
2. Commit.

**Expected behavior**

It should fail because it created new files. That means the template output is out of sync, and thus there's a hook failure.

**Additional Information**

- `copier_template_tester` Version:  1.0.2
- OS: \[e.g. macOS/Windows\]: Linux
- OS Version \[e.g. Catalina/10\]: Fedora 37

**Comment by KyleKing** [comment]
I can replicate the issue like this:

```sh
~/Developer/packages/calcipy_template  main≡ > nvim "asdf"ctt.toml
~/Developer/packages/calcipy_template  main≡  ~2 > git add -A
~/Developer/packages/calcipy_template  main≡  ~2 > git commit -m "asdf"

copier-template-tester...................................................Passed
check for added large files..............................................Passed
check that executables have shebangs.................(no files to check)Skipped
check json...........................................(no files to check)Skipped
check for merge conflicts................................................Passed
check for broken symlinks............................(no files to check)Skipped
check toml...............................................................Passed
check vcs permalinks.....................................................Passed
check yaml...............................................................Passed
debug statements (python)............................(no files to check)Skipped
detect destroyed symlinks................................................Passed
detect private key.......................................................Passed
fix double quoted strings............................(no files to check)Skipped
fix end of files.........................................................Passed
fix utf-8 byte order marker..............................................Passed
fix python encoding pragma...........................(no files to check)Skipped
forbid new submodules................................(no files to check)Skipped
mixed line ending........................................................Passed
pretty format json...................................(no files to check)Skipped
trim trailing whitespace.................................................Passed
beautysh.............................................(no files to check)Skipped
mdformat.............................................(no files to check)Skipped
pyproject-fmt........................................(no files to check)Skipped
shellcheck...........................................(no files to check)Skipped
Detect secrets...........................................................Passed
yamlfix..............................................(no files to check)Skipped
copier_forbidden_files...............................(no files to check)Skipped
[main 8a440e8] asdf
 3 files changed, 6 insertions(+), 2 deletions(-)

~/Developer/packages/calcipy_template  main↑1  ?1 > git status
On branch main
Your branch is ahead of 'origin/main' by 1 commit.
  (use "git push" to publish your local commits)

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .ctt/new/

nothing added to commit but untracked files present (use "git add" to track)
~/Developer/packages/calcipy_template  main↑1  ?1 >
```

I think to fix this, `ctt` would need to check if any of the folders were untracked and `exit 1`, but otherwise defer to `pre-commit` to check for differences on tracked files.

I don't think this is a `pre-commit` bug because `pre-commit` intentionally ignores untracked files: https://github.com/pre-commit/pre-commit/issues/708#issuecomment-368245812

**Comment by KyleKing** [comment]
Hi @yajo, thanks for following up. Sorry to have let this issues sit stale for so long

I think I have this issue fixed with v1.1.0. Feel free to reopen this issue if `ctt` isn't worked as expected!
